ddtrace/writer.py

```diff
@@ -120,6 +126,13 @@ def _target(self):
         while True:
             traces = self._trace_queue.pop()
             if traces:
+                # Before sending the traces, make them go through the
+                # processing pipeline
+                try:
+                    traces = self._apply_processing_pipeline(traces)
+                except Exception as err:
+                    log.error("error while processing traces:{0}".format(err))
+            if traces:
```

Comment:
I kept two `if traces:` because `self._apply_processing_pipeline` may return no traces if all of them are filtered out by the pipeline.